cfgov/unprocessed/apps/analytics-gtm/js/bah-closing-disclosure-listeners.js

```diff
@@ -1,68 +1,145 @@
-// TODO: Remove $.
-import $ from 'jquery';
-
 import {
-  delay,
+  addEventListenerToElems,
+  closest,
   track
 } from './util/analytics-util';
 
-const OAHCDAnalytics = ( function( $ ) {
+// Owning a Home /owning-a-home/closing-disclosure/
+
+const OAHCDAnalytics = ( () => {
+
+  const expandableStates = {};
+  let lastExpandable;
+
+  /**
+   * Record state (expanded or collapsed) of expandables.
+   * @param {string} id - The unique HTML ID of an expandable.
+   */
+  function recordExpandableState( id ) {
+    if ( !expandableStates[id] ) {
+      expandableStates[id] = true;
+    } else if ( lastExpandable === id ) {
+      expandableStates[id] = !expandableStates[id];
+    }
+    lastExpandable = id;
+  }
+
+  /**
+   * Record state (expanded or collapsed) of expandables.
+   * @param {string} expandable - An expandable HTML element.
+   * @returns {boolean} True if the expandable is animating, false otherwise.
+   */
+  function isAnimatingExpandable( expandable ) {
+    let isAnimating = false;
+
+    if ( expandable.classList.contains( 'o-expandable__expanding' ) ||
+         expandable.classList.contains( 'o-expandable__collapsing' ) ) {
+      isAnimating = true;
+    }
 
-  $( '.tab-link' ).click( function() {
-    const text = $( this ).text().trim();
+    return isAnimating;
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackTabLinkClick( event ) {
+    const target = event.target;
+    const text = target.textContent.trim();
     track( 'OAH Closing Disclosure Interaction', 'Tab click', text );
-  } );
+  }
 
-  $( '.form-explainer_page-link' ).click( function() {
-    const pageNumber = 'Page ' + $( this ).attr( 'data-page' );
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackFormExplainerPageLinkClick( event ) {
+    const target = event.target;
+    const pageNumber = 'Page ' + target.getAttribute( 'data-page' );
     track(
       'OAH Closing Disclosure Interaction', 'Page link click', pageNumber
     );
-  } );
+  }
 
-  $( '.form-explainer_page-buttons button' ).click( function() {
-    const currentPage = 'Page ' + $( '.form-explainer_page-link.current-page' ).attr( 'data-page' );
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackFormExplainerPageButtonClick( event ) {
+    const target = event.currentTarget;
+    const currentPageDom =
+      document.querySelector( '.form-explainer_page-link.current-page' );
+    const currentPage = 'Page ' + currentPageDom.getAttribute( 'data-page' );
     let action = 'Next Page button clicked';
-    if ( $( this ).hasClass( 'prev' ) ) {
+    if ( target.classList.contains( 'prev' ) ) {
       action = 'Previous Page button clicked';
     }
     track( 'OAH Closing Disclosure Interaction', action, currentPage );
+  }
+
+  /**
+   * @param {MouseEvent} event - Mouse event from the click.
+   */
+  function trackExpandableTargetsClick( event ) {
```

Comment:
Instead of tracking the click and checking the state, would it better to watch for DOM changes with a MutationObserver?

Reply:
The analytics should support a much wider net than what we use elsewhere and not necessitate polyfills. MutationObserver does throw errors in New Relic once in awhile, though that may mean the expandables aren't working anyway. As mentioned in another comment I think what we want is the architecture to communicate between components so that analytics can just observe what events are being emitted instead of snooping in the dom.

![screen shot 2018-05-04 at 1 09 40 pm](https://user-images.githubusercontent.com/704760/39641611-d76e3f9a-4f9c-11e8-9c83-72278ab93a7e.png)